reactor-core/src/main/java/reactor/core/publisher/FluxBufferTimeout.java

```diff
@@ -155,6 +161,12 @@ protected void doOnSubscribe() {
 
 		void nextCallback(T value) {
 			synchronized (this) {
+				if (OUTSTANDING.decrementAndGet(this) < 0)
+				{
+					actual.onError(Exceptions.failWithOverflow("Unrequested element received"));
```

Comment:
you will need to apply discarding here. Since you directly call `actual.onError`, you'd need discarding of both the current buffer and the un-added value:
```java
Context ctx = actual.currentContext();
Operators.onDiscard(value, ctx);
Operators.onDiscardMultiple(values, ctx);
```

Reply:
Thanks @simonbasle - I've implemented this change and it fixed the failing test. Not terribly surprising since the test is of the onDiscard!